### PR TITLE
Add keystone options to client

### PR DESCRIPTION
**Pull request by rbrndt**
If keystone options are not configured, will
maintain current behavior

### CODE REVIEW

File: monascaclient/keystone.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+)
+
+type KeystoneConfig gophercloud.AuthOptions
+
+func SetKeystoneToken() {
+	monClient.setKeystoneToken()
+}
+
+func (c *Client) setKeystoneToken() error {
+	openstackProvider, err := openstack.AuthenticatedClient(c.keystoneConfig)
```
Comment (mhoppal) [inline monascaclient/keystone.go]: awesome work on this!!! can we have the option to read from env as well?

Comment (mhoppal) [reply monascaclient/keystone.go]: https://github.com/monasca/smoke-test/blob/master/smoke-test.go#L205

File: monascaclient/client.go [hunk 2]
```diff
@@ -137,12 +146,28 @@ func (c *Client) callMonasca(monascaURL string, method string, requestBody *[]by
...
+		c.setKeystoneToken()
+		c.applyHeaders(req)
+		resp, respErr = client.Do(req)
 	}
 
-	return resp, err
+	return resp, respErr
+}
+
+func (c *Client) applyHeaders(req http.Request) {
```
Comment (mhoppal) [inline monascaclient/client.go]: pass by value?

**Review by mhoppal — APPROVED** [review]
👍  testing once wifi decides to work and will merge if it works

File: monascaclient/keystone.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	"net/http"
+)
+
+type KeystoneConfig gophercloud.AuthOptions
+
+func SetKeystoneToken() error {
+	return monClient.setKeystoneToken()
+}
+
+func (c *Client) setKeystoneToken() error {
```
Comment (craigbr) [inline monascaclient/keystone.go]: Personally, I would rather this just returned the token and the code in client.go did things like headers.Add(). Seems strange to call this and have it overwrite the headers in the client

File: monascaclient/keystone.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	openstackProvider, err := openstack.AuthenticatedClient(c.keystoneConfig)
+	if err != nil {
+		return err
+	}
+	//fmt.Println(openstackProvider.TokenID)
```
Comment (craigbr) [inline monascaclient/keystone.go]: Remove

File: monascaclient/keystone.go (hunk 3 above)
Comment (rbrndt) [reply monascaclient/keystone.go]: Where in the client.go did you think this should go? I don't see an obvious place.

**Pull request merged**